bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -738,6 +627,159 @@ public Image(Device device, ImageDataProvider imageDataProvider) {
 	init();
 }
 
+
+/**
+ * Upon zoom update, this method copies the image data and add disabled or grayed image depending on the flag
+ *
+ * @param device the device on which to create the image
+ * @param data the image data to create the image from (must not be null)
+ * @param flag the style, either <code>IMAGE_DISABLE</code> or <code>IMAGE_GRAY</code>
+ * @return new image data with valid styling properties as per the flag
+ */
+private ImageData copyImageDataForDisabledOrGrayImages(Device device, ImageData data, int flag) {
```

Comment:
Why does this method take a device and a flag? Can't this information be taken from the image (`this`) itself? For the device parameter, I see no caller using another value than the device of the image, and for the flag parameter it is only one case where `SWT.Default` is passed explicitly, even though the image's default `SWT.IMAGE_COPY` should lead to the same result here.

Reply:
I passed device and a flag because it could be from another object of an image, the win32_getHandle method passes different object of the Image which we get from GC. 

For the second part, initially I did use the SWT.IMAGE_COPY as you suggested but andy had a good point in his comment above that we are not actually copying anything and this flag is kind of gives different idea.

Reply:
Maybe I miss something, but the calls of the method I see always looks like this: `image.copyImageDataForDisabledOrGrayImages(image.getDevice(), resizedData, image.styleFlag)`
So you extract the device and the flags out of the image on which you call the method. Only difference is the `initNative` method, but there the device is also retrieved from `this` and the flag is always retrieved from the image (i.e., `this`) by the caller, expect for one caller using `SWT.DEFAULT`, but that could also simply reduce to the default flag of the image begin `SWT.IMAGE_COPY`. So I do not understand why there parameter are necessary at all and why simply using `this.getDevice()` and `this.styleFlag` is not sufficient.

Reply:
Now I get your point, you're right. We could just use the `this` since it was called by that image instances.